src/triangle.rs

```diff
@@ -0,0 +1,343 @@
+    #[inline]
+    pub fn is_nan(&self) -> bool {
+        self.a.is_nan() || self.b.is_nan() || self.c.is_nan()
+    }
+
+    /// [`Triangle`]'s vertices as an array of Points
+    #[inline]
+    pub fn as_array(&self) -> [Point; 3] {
+        [self.a, self.b, self.c]
+    }
+}
+
+impl From<(Point, Point, Point)> for Triangle {
+    fn from(points: (Point, Point, Point)) -> Triangle {
+        Triangle::new(points.0, points.1, points.2)
+    }
+}
+
+impl Add<Vec2> for Triangle {
+    type Output = Triangle;
+
+    #[inline]
+    fn add(self, v: Vec2) -> Triangle {
+        Triangle::new(self.a + v, self.b + v, self.c + v)
+    }
+}
+
+impl Sub<Vec2> for Triangle {
+    type Output = Triangle;
+
+    #[inline]
+    fn sub(self, v: Vec2) -> Triangle {
+        Triangle::new(self.a - v, self.b - v, self.c - v)
+    }
+}
+
+#[doc(hidden)]
+pub struct TrianglePathIter {
+    triangle: Triangle,
+    ix: usize,
+}
+
+impl Shape for Triangle {
+    type PathElementsIter<'iter> = TrianglePathIter;
+
+    fn path_elements(&self, _tolerance: f64) -> TrianglePathIter {
+        TrianglePathIter {
+            triangle: *self,
+            ix: 0,
+        }
+    }
+
+    #[inline]
+    fn area(&self) -> f64 {
+        Triangle::area(self)
+    }
+
+    #[inline]
+    fn perimeter(&self, _accuracy: f64) -> f64 {
+        self.a.distance(self.b) + self.b.distance(self.c) + self.c.distance(self.a)
+    }
+
+    #[inline]
+    fn winding(&self, pt: Point) -> i32 {
+        let s0 = (self.b - self.a).cross(pt - self.a).signum();
+        let s1 = (self.c - self.b).cross(pt - self.b).signum();
+        let s2 = (self.a - self.c).cross(pt - self.c).signum();
+
+        if s0 == s1 && s1 == s2 {
+            s0 as i32
+        } else {
+            0
+        }
+    }
+
+    #[inline]
+    fn bounding_box(&self) -> Rect {
+        Rect::new(
+            self.a.x.min(self.b.x.min(self.c.x)),
+            self.a.y.min(self.b.y.min(self.c.y)),
+            self.a.x.max(self.b.x.max(self.c.x)),
+            self.a.y.max(self.b.y.max(self.c.y)),
+        )
+    }
+
+    #[inline]
+    fn as_rect(&self) -> Option<Rect> {
+        Some(self.bounding_box())
+    }
+}
+
+// NOTE: vertices a, b and c are not guaranteed to be in order as described in the struct comments
+//       (i.e. as "vertex a is topmost, vertex b is leftmost, and vertex c is rightmost")
+impl Iterator for TrianglePathIter {
+    type Item = PathEl;
+
+    fn next(&mut self) -> Option<PathEl> {
+        self.ix += 1;
+        match self.ix {
+            1 => Some(PathEl::MoveTo(self.triangle.a)),
+            2 => Some(PathEl::LineTo(self.triangle.b)),
+            3 => Some(PathEl::LineTo(self.triangle.c)),
+            4 => Some(PathEl::ClosePath),
+            _ => None,
+        }
+    }
+}
+
+// TODO: better and more tests
+#[cfg(test)]
+mod tests {
+    use crate::{Point, Triangle, Vec2};
+
+    fn assert_approx_eq(x: f64, y: f64) {
+        assert!((x - y).abs() < 1e-7);
```

Comment:
This is quite a loose bound for `f64`.  I'd use something closer to machine epsilon e.g. `1e-13`. Out of scope for this PR, but we really need to standardize this (I like [`float_cmp`](https://docs.rs/float-cmp/latest/float_cmp/) because it allows big epsilons as long as the expected and actually are <n representable numbers apart in f64).

Reply:
I copied the bound from "rect.rs" i understood that it was loose but knew that it was only a temporary solution. 

i could change it to be more strict but even with `1e-13`, `Triangle::area()` tests are already failing from  precision. i still could arbitrarily restrict the bound or could just wait for it to be standardized as you said, which would you prefer?

Reply:
Just leave it as-is for now - what you have is fine.